sky/serve/load_balancer.py

```diff
@@ -70,30 +79,121 @@ def _sync_with_controller(self):
                     ready_replica_urls = response.json().get(
                         'ready_replica_urls')
                 except requests.RequestException as e:
-                    print(f'An error occurred: {e}')
+                    logger.error('An error occurred when syncing with '
+                                 f'the controller: {e}')
                 else:
                     logger.info(f'Available Replica URLs: {ready_replica_urls}')
-                    self._load_balancing_policy.set_ready_replicas(
-                        ready_replica_urls)
-            time.sleep(constants.LB_CONTROLLER_SYNC_INTERVAL_SECONDS)
+                    with self._client_pool_lock:
+                        self._load_balancing_policy.set_ready_replicas(
+                            ready_replica_urls)
+                        for replica_url in ready_replica_urls:
+                            if replica_url not in self._client_pool:
+                                self._client_pool[replica_url] = (
+                                    httpx.AsyncClient(base_url=replica_url))
+                        urls_to_close = set(
+                            self._client_pool.keys()) - set(ready_replica_urls)
+                        client_to_close = []
+                        for replica_url in urls_to_close:
+                            client_to_close.append(
+                                self._client_pool.pop(replica_url))
+                    for client in client_to_close:
+                        asyncio.run(client.aclose())
```

Comment:
profile the performance?

Reply:
I benchmarked time to close a client and it is in the order of milliseconds, mostly ranging from 2 to 6 ms. I also change it to async execution to run it in the background for a safeguard.